app/Utils/UrlGenerator.php

```diff
@@ -33,4 +33,85 @@ public static function asset_url($fileType, $filename)
 
         return base_url($filesPath . $filename);
     }
+
+    public static function generate_og_image($originalImageRoute)
+    {
+        $originalImagePath = base_url() . ltrim($originalImageRoute, '/');
+
+        $ogImageWidth  = 1200;
+        $ogImageHeight = 630;
+
+        $pathInfo              = pathinfo($originalImageRoute);
+        $originalFileName      = $pathInfo['filename'];
+        $originalFileExtension = $pathInfo['extension'];
+        $newFileName           = $originalFileName . '-metaimage.png';
+        $newFilePath           = $pathInfo['dirname'] . '/' . $newFileName;
+
+        if (! file_exists($newFilePath)) {
+            if (str_ends_with($originalImagePath, '.png')) {
+                $originalImage = imagecreatefrompng($originalImagePath);
+            } else {
+                $originalImage = imagecreatefromjpeg($originalImagePath);
+            }
+
+            $originalWidth  = imagesx($originalImage);
+            $originalHeight = imagesy($originalImage);
+
+            if ($originalWidth < $ogImageWidth && $originalHeight < $ogImageHeight) {
```

Comment:
No deberia ser un or esto?
```suggestion
            if ($originalWidth < $ogImageWidth || $originalHeight < $ogImageHeight) {
```

Reply:
No because if it is bigger on width or height it needs to adapt differently than if the image is smaller in both width and height